### PR TITLE
add mock

### PR DESCRIPTION
**Pull request by s77rt**
cc @tomekzaw 

### Related Issues

https://github.com/Expensify/App/issues/52475

### CODE REVIEW

File: src/MarkdownTextInput.tsx [hunk 1]
```diff
@@ -28,10 +28,9 @@ function initializeLiveMarkdownIfNeeded() {
   if (initialized) {
     return;
   }
-  if (!NativeLiveMarkdownModule) {
-    throw new Error('[react-native-live-markdown] NativeLiveMarkdownModule is not available');
+  if (NativeLiveMarkdownModule) {
+    NativeLiveMarkdownModule.install();
```
Comment (tomekzaw) [inline src/MarkdownTextInput.tsx:32]: What's the reason for this change? If `NativeLiveMarkdownModule` is not injected properly, we'd like to fail immediately instead of showing a non-actionable error "global.jsi_setMarkdownRuntime is not available" that sounds more like an actual implementation bug rather than setup issue.

File: tsconfig.json [hunk 2]
```diff
@@ -26,6 +26,6 @@
     "verbatimModuleSyntax": true,
     "typeRoots": ["node_modules/@types"]
   },
-  "include": ["src/**/*"],
+  "include": ["src/**/*", "mock/**/*"],
```
Comment (tomekzaw) [inline tsconfig.json:29]: Can we move `mock` directory to `src/` directory? Or would this affect the paths in imports? Let's avoid `@expensify/react-native-live-markdown/src/mock` if possible (with `src/mock`)

File: src/MarkdownTextInput.tsx (hunk 1 above)
Comment (s77rt) [reply src/MarkdownTextInput.tsx:32]: This is required for the mock to work. Otherwise we'd have to [mock react-native TurboModuleRegistry](https://github.com/Expensify/App/blob/4a9bd7cb3bbbbd95ec7b9a229611786ebe4dbbe8/__mocks__/react-native.ts#L45-L59).

File: tsconfig.json (hunk 2 above)
Comment (s77rt) [reply tsconfig.json:29]: It will affect the import path indeed. It's better to keep the path as `@expensify/react-native-live-markdown/mock`

**Review by Kicu — APPROVED** [review]
I think the mock itself looks good.

I'm not sure about the `NativeLiveMarkdownModule.install()` part because I don't have much experience interacting with `TurboModuleRegistry`.
Everything else seems reasonable to me 👍

**Review by tomekzaw — APPROVED** [review]
LGTM, also tested all example apps and everything looks good. Thanks for the PR!
